concurrent/src/main/java/io/airlift/concurrent/VmRuntimeTicker.java

```diff
@@ -0,0 +1,180 @@
+package io.airlift.concurrent;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Supplier;
+import com.google.common.base.Suppliers;
+import com.google.common.base.Ticker;
+import io.airlift.log.Logger;
+import io.airlift.stats.TimeStat;
+import io.airlift.units.Duration;
+import org.weakref.jmx.Managed;
+
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicLong;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+import static java.util.concurrent.TimeUnit.NANOSECONDS;
+
+/**
+ * A ticker that does not advance when the VM is paused.
+ *
+ * This is implemented using a simple background thread that sleeps for a small
+ * period and then measures the sleep duration.  If the duration is small also,
+ * the ticker is advanced; otherwise it is ignored.
+ *
+ * The pause durations are recorded in a TimeStat for monitoring.
+ */
+public class VmRuntimeTicker
+        extends Ticker
```

Comment:
This original point of this code was to us this ticker to measure durations that ignore vm pauses.  So you could say, I'm going to try something for 30 seconds regardless of VM pauses.